### ISSUE TITLE
Incorrect diagnostic that Deconstruct method is unused.

### ISSUE THREAD

**Issue by gafter** [dotnet/roslyn#75995]
Roslyn reports that `Deconstruct` is unused in the C# code below. But it is used. You would get an error if you remove the method.

```csharp
class C
{
    void M(
        ref object o,
        ref object p)
    {
        (o, p) = this;
    }
    
    void Deconstruct(
        out object? o,
        out object? p)
    {
        o = null;
        p = null;
    }
}
```

**Comment by CyrusNajmabadi** [comment]
@gafter are you getting a compiler warning, or an analyzer warning?  If so, can you include the diagnostic id?

**Comment by gafter** [comment]
I don't know. See [https://sharplab.io/#v2:CYLg1APgAgTABAYQLACgDeq5blALHAWQApNsyAnAUwDM4B7AIwCtKBjAF3oBpSysrajFhzgAHAJS84GFH2xE6XMeLgBeOOwAWASwDOAbikBfKVLxwAImzoA7Xe3IBXDiVlz6jzkLbsA/NykyOk96Zh9/CSkZd3o1OBtHABtEwzc5UTiE5NSyExQ8oA==](https://sharplab.io/#v2:CYLg1APgAgTABAYQLACgDeq5blALHAWQApNsyAnAUwDM4B7AIwCtKBjAF3oBpSysrajFhzgAHAJS84GFH2xE6XMeLgBeOOwAWASwDOAbikBfKVLxwAImzoA7Xe3IBXDiVlz6jzkLbsA/NykyOk96Zh9/CSkZd3o1OBtHABtEwzc5UTiE5NSyExQ8oA==)

**Comment by CyrusNajmabadi** [comment]
@jcouv I can't figure out how to determine that the .Deconstruct method is used here.  How is this exposed in teh semantic model?  Also checking with @333fred .  I do see a IDeconstructAssignmentOperation, but it seems to be missing the target method, unless i'm missing something.  Could you guys advise?  How do you encode this into the tree?
